core/src/avm1/activation.rs

```diff
@@ -316,4 +337,33 @@ impl<'gc> Activation<'gc> {
             }
         }
     }
+
+    /// Attempts to lock the activation frame for execution.
+    ///
+    /// If this frame is already executing, that is an error condition that we
+    /// panic out of.
+    pub fn lock_or_panic(&mut self) {
```

Comment:
Does this have to be "or_panic"? Can't we return if it successfully locked and allow something else to deal with the failure? Even if that's temporarily panicking until we have nice error management, I don't like this being baked in.

Reply:
I think this was added in before I made native functions falliable - it totally can return Result<(), Error> now